simpleGraph/dynamicAln.go

```diff
@@ -0,0 +1,211 @@
+package simpleGraph
+
+import (
+	"github.com/edotau/simpleio"
+	"github.com/vertgenlab/gonomics/common"
+	"github.com/vertgenlab/gonomics/dna"
+	"log"
+)
+
+func RightDynamicAln(alpha []dna.Base, beta []dna.Base, scores [][]int64, gapPen int64, m [][]int64, trace [][]simpleio.CigarOp) (int64, []simpleio.ByteCigar, int, int, int, int) {
+	//check if size of alpha is larger than m
+	var currMax int64
+	var maxI int
+	var maxJ int
+	var i, j, routeIdx int
+	//setting up the first rows and columns
+	//seting up the rest of the matrix
+	for i = 0; i < len(alpha)+1; i++ {
+		for j = 0; j < len(beta)+1; j++ {
+			if i == 0 && j == 0 {
+				m[i][j] = 0
+			} else if i == 0 {
+				m[i][j] = m[i][j-1] + gapPen
+				trace[i][j] = 'I'
+			} else if j == 0 {
+				m[i][j] = m[i-1][j] + gapPen
+				trace[i][j] = 'D'
+			} else {
+				m[i][j], trace[i][j] = simpleio.ByteMatrixTrace(m[i-1][j-1]+scores[alpha[i-1]][beta[j-1]], m[i][j-1]+gapPen, m[i-1][j]+gapPen)
```

Comment:
hard to say for sure, because I know I have not seen the simpleio package, but it seems like maybe functions deep in the alignment code should be with the other alignment code?  From the name I am guessing that simpleio deals mostly with input and output and making it simpler?